store/src/java/com/zimbra/cs/service/mail/SendShareNotification.java

```diff
@@ -745,5 +765,83 @@ public void run() {
+        if (Provisioning.onLocalServer(acct)) {
+            Mailbox mbox = MailboxManager.getInstance().getMailboxByAccountId(granteeId);
+            // if file is shared, this file should be available at receivers end
+            if (reqItem.getType().equals(Type.DOCUMENT) && action == null) {
+                Document docReqItem = (Document) reqItem;
+                mbox.createFileSharedWithMe(mbox, octxt, FolderConstants.ID_FOLDER_FILE_SHARED_WITH_ME, fileName,
+                        ownerAccountId, ownerFileId, fileUUID, fileOwnerName, docReqItem.getContentType(),
+                        docReqItem.getSize(), rights);
+            } // if file access is revoked at source delete it from receiver end as well
+            else if (reqItem.getType().equals(Type.DOCUMENT) && action != null && action.name().equals(REVOKE)) {
+                String fileUUIDToDelete = fileUUID + ":" + ownerFileId;
+                mbox.deleteFileSharedWithMe(mbox, octxt, fileUUIDToDelete);
+            } // if file rights is changed , update file permission accordingly
+            else if (reqItem.getType().equals(Type.DOCUMENT) && action != null
+                    && action.name().equalsIgnoreCase(EDIT)) {
+                Document docReqItem = (Document) reqItem;
+                mbox.updateFileSharedWithMe(mbox, octxt, ownerAccountId, ownerFileId, fileUUID, fileOwnerName,
+                        docReqItem.getContentType(), rights);
+            }
+        } else {
+            Document docReqItem = (Document) reqItem;
+            FileSharedWithMeRequest req = new FileSharedWithMeRequest();
+            req.setAction(action == null ? CREATE : action.name());
+            req.setGranteeId(granteeId);
+            req.setFileName(fileName);
+            req.setOwnerAccountId(ownerAccountId);
+            req.setOwnerFileId(ownerFileId);
+            req.setFileUUID(fileUUID);
+            req.setFileOwnerName(fileOwnerName);
+            req.setRights(rights);
+            req.setContentType(docReqItem.getContentType());
+            req.setSize(docReqItem.getTotalSize());
+
+            Element request = JaxbUtil.jaxbToElement(req);
+            sendRequest(acct, request, null, true, ZIMBRA_SOAP_TIMEOUT);
+        }
+    }
+
+    static Element sendRequest(Account acct, Element request, String uri, boolean authRequired, int timeout)
+            throws ServiceException {
+        assert request != null;
+        assert acct != null;
 
+        String soapUri = null;
+        if (uri == null || uri.length() == 0) {
+            soapUri = AccountUtil.getSoapUri(acct);
+        } else {
+            soapUri = uri + AccountConstants.USER_SERVICE_URI;
+        }
+        SoapHttpTransport transport = new SoapHttpTransport(soapUri);
+        try {
+            transport.setUserAgent(ZIMBRA_SOAP_USER_AGENT, ZIMBRA_SOAP_VERSION);
+            transport.setTimeout(timeout);
+            transport.setRetryCount(1);
```

Comment:
Do we need to set them rather than use default values set in SoapHttpTransport?

Reply:
updated it to get data from SoapHttpTransport class

Reply:
the method is removed and used proxyRequest